app/src/main/java/ch/epfl/sweng/studyup/question/QuestionParser.java

```diff
@@ -49,44 +49,42 @@
             Uri imageUri;
             boolean trueFalse;
             int answerNumber;
-
             while ((line = bufferedReader.readLine()) != null) {
                 File nf = new File(line);
-                if(checkIfFileExists && !nf.exists()){throw new FileNotFoundException("The image for the question has not been found"); }
+                if (checkIfFileExists && !nf.exists()) {
+                    bufferedReader.close();
+                    isr.close();
+                    throw new FileNotFoundException("The image for the question has not been found");
+                }
                 imageUri = Uri.parse(line);
 
                 if ((line = bufferedReader.readLine()) == null) {
-                    Log.e(TAG, "While reading the file: second line is empty");
+                    bufferedReader.close();
+                    isr.close();
+                    Log.e(TAG, "While reading the questions file: second line is empty");
                     toast.show();
                     return null;
                 }
                 trueFalse = Boolean.parseBoolean(line);
 
                 if ((line = bufferedReader.readLine()) == null) {
-                    Log.e(TAG, "While reading the file: third line is empty");
+                    bufferedReader.close();
+                    isr.close();
+                    Log.e(TAG, "While reading the questions  file: third line is empty");
                     toast.show();
                     return null;
                 }
                 answerNumber = Integer.parseInt(line);
 
                 list.add(new Question(imageUri, trueFalse, answerNumber));
             }
-
-        }catch (UnsupportedEncodingException e) {
-            Log.e(TAG, "Unsupported encoding: \n" + e.getMessage());
-
-            /*
-                Version with stackTrace:
-                Log.e(TAG, "Unsupported encoding: \n");
-                for(StackTraceElement elem: e.getStackTrace()) {
-                    System.err.println("-File:"+elem.getClassName()+" -Method:L"+elem.getMethodName()+" -Line:"+elem.getLineNumber());
-                }
-            */
-
-            toast.show();
-            return null;
+        }catch (FileNotFoundException e) {
+            throw e;
```

Comment:
In my opinion, we shouldn't throw an error if we can avoid throwing it. But if it is really throwing it that you want, we could just not catch it right?

Reply:
Yes I can avoid throwing it. But there's another method that can throw an IOException and if I don't catch the FileNotFoundException before it gets caught by the IOException.

Reply:
Maybe we could replace it with a Toast